cla_frontend/assets-src/javascripts/app/js/controllers/call_centre/assign_provider.js

```diff
@@ -64,12 +65,14 @@
 
           $scope.case.$set_matter_types().then(
             function () {
+              $scope.form_is_submitting = true;
               $scope.case.$assign(data).then(
                 function(response) {
                   $state.go('case_list');
                   flash('success', 'Case ' + $scope.case.reference + ' assigned to ' + response.data.name);
                 },
                 function(data) {
+                  $scope.form_is_submitting = false
```

Comment:
Do we need this line? It is set to false by default, so if the form submission fails it won't do anything anyway

Reply:
Think about what you'd want to happen if someone submitted an invalid form (I think that's what this is about, anyway; Said will correct me if I'm wrong)

Reply:
Ahh my bad `form_is_submitting` gets set to true as soon as they click on the button. I thought it was getting set after the response. Thanks

Reply:
@mayowaadeniyi yes I think it's need.
You are right, it is set to false by default but on line 68 I set `$scope.form_is_submitting` to true.
When there is a form submission error, the button needs to be enabled again.

Although your comment made me questioned whether `form_is_submitting` is appropriate name for what is being used for? Maybe I should change it to something like `case_assignment_in_progress` or something related to case assignment, what do you think?

Reply:
I think that's a good point. `case_assignment_in_progress` sounds like a good name to me. It's in the assign provider file, so I was thinking something like `assigning_provider_status`

Reply:
I would prefer it if it was either a true/false variable. status indicates it would have multiple values like ready, in-progress and complete, which might be a bit of an overkill for this.

So i think i'll go with `assigning_provider_in_progress` as the variable name.